util/src/main/java/com/google/cloud/hadoop/util/logging/CustomLoggingProvider.java

```diff
@@ -0,0 +1,19 @@
+package com.google.cloud.hadoop.util.logging;
+
+public abstract class CustomLoggingProvider {
+    private static CustomLoggingProvider customLoggingProvider = null;
+
+    public static synchronized void setCustomLoggingProvider(CustomLoggingProvider provider) {
+        customLoggingProvider = provider;
+    }
+
+    public static CustomLoggingProvider getInstance() {
+        if (customLoggingProvider != null)
+            return customLoggingProvider;
+
+        customLoggingProvider = new DefaultLoggingProvider();
+        return customLoggingProvider;
+    }
+
+    public abstract void log(String logMessage);
```

Comment:
An alternative to consider would be:
`public abstract void log(FSOpType type, FSOpStatus status, String path, String format, String... params);`
where FSOpType is an enum with values such as CREATE, DELETE, RENAME etc. FSOpStatus is an enum with values such as STARTED, SUCCEEDED, FAILED, UNDEFINED.
an example usage could be: 
`CustomLoggingProvider.getInstance().log(FSOpType.CREATE, FSOpStatus.SUCCEEDED, finalGcsPath.toString(), "size: {}, time: {}", statistics.getBytesWritten(), statistics.getTime());`
This would give is more structured information to deal with downstream.